xla/service/cpu/onednn_layer_norm.cc

```diff
@@ -57,15 +58,11 @@ ABSL_ATTRIBUTE_NO_SANITIZE_MEMORY void __xla_cpu_runtime_OneDnnLayerNorm(
       static_cast<const xla::ExecutableRunOptions*>(args[arg_indx++]);
   XLA_LIGHTWEIGHT_CHECK(run_options != nullptr);
   XLA_LIGHTWEIGHT_CHECK(run_options->intra_op_thread_pool() != nullptr);
-  tsl::OneDnnThreadPool thread_pool(
-      run_options->intra_op_thread_pool()->getPool(), false);
-  engine cpu_engine(engine::kind::cpu, 0);
-#ifndef ENABLE_ONEDNN_OPENMP
-  auto onednn_stream =
-      stream(dnnl::threadpool_interop::make_stream(cpu_engine, &thread_pool));
-#else
-  auto onednn_stream = stream(cpu_engine);
-#endif  // ENABLE_ONEDNN_OPENMP
+  auto thread_pool = CreateOneDnnThreadPool(
```

Comment:
Not for this CL, but in general instead of defining onednn primitives as custom calls here, they should be defined as "normal" c++ APIs under runtime/onednn and called directly from thunk(s).

We don't want to keep functions with this ugly C API for long, and especially supressing ASAN unconditionally. With thunks asan/msan/tsan should just work.